atom_gems/AtomTutorials/Templates/VegetationBending/MeshMotionVectorVegetationBending/MeshMotionVectorVegetationBending.azsl

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/Features/PBR/DefaultObjectSrg.azsli>
+#include <VegetationBending_Common.azsli>
+
+#include <scenesrg.srgi>
+#include <viewsrg.srgi>
+
+#include <Atom/RPI/ShaderResourceGroups/DefaultDrawSrg.azsli>
+
+struct VSInput
+{
+    float3 m_position : POSITION;
+    float3 m_normal : NORMAL;
+
+    float4 m_optional_color : COLOR0;
+};
+
+struct VSOutput
+{
+    float4 m_position : SV_Position;
+    float3 m_worldPos : TEXCOORD0;
+    float3 m_worldPosPrev: TEXCOORD1;
+};
+
+struct PSOutput
+{
+    float2 m_motion : SV_Target0;
+};
+
+VSOutput MainVS(VSInput IN)
+{
+    VSOutput OUT;
+    
+    float4x4 objectToWorld = SceneSrg::GetObjectToWorldMatrix(ObjectSrg::m_objectId);
```

Comment:
I'm getting a bit of deja vu here, how is this file different from the other motion vector shader?

Reply:
This folder is a 'Templates' folder, and the files in it represent the starting point for the tutorial. Copy-paste this file, follow tutorial steps, and you should end up with the 'final' version. The difference is that this one is missing a call to ProcessBending with the previous frame's time in order to get the previous frame's position.

It does seem to split the difference though, where this starting point is neither the original motion vector shader nor the one of the original Vegetation vertex shaders. I think it would be better to replace this with the shader from MeshMotionVectorCommon.azsli, and then show how to go from there to the final version.

Reply:
Gotcha the directory got cut off so I couldn't see the full file paths, that makes sense thanks! Agreed that maybe some of the implementation in your template files should be left "as exercises to the reader" so to speak for tutorialization.

Reply:
True, I originally wanted to allow users to just pick up the motion vector shader and not repeat steps that they had done before, but I can see how that can lead to this confusing code. I can't quite copy the MeshMotionVectorCommon.azsli shader because they have a previous position parameter that this one doesn't have, but I updated the template motion vector shader to delete the first call to ProcessBending.